### PR TITLE
add ReplicatedMap interface with GoDocs

### PR DESCRIPTION
**Pull request by furkansenharputlu**
#158

### CODE REVIEW

File: core/replicated_map.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+package core
+
+import (
+	. "github.com/hazelcast/hazelcast-go-client/serialization"
+	"time"
+)
+
+// ReplicatedMap is a map-like data structure with weak consistency
+// and values locally stored on every node of the cluster.
+//
```
Comment (SaitTalhaNisanci) [inline core/replicated_map.go]: There shouldnt be any space here otherwise it wont be generated with goDoc

File: core/replicated_map.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+// Whenever a value is written asynchronously, the new value will be internally
+// distributed to all existing cluster members, and eventually every node will have
+// the new value.
+//
```
Comment (SaitTalhaNisanci) [inline core/replicated_map.go]: should remove the space

File: core/replicated_map.go (hunk 1 above)
Comment (furkansenharputlu) [reply core/replicated_map.go]: No, it generates. The space is used for new paragraph.

File: core/replicated_map.go (hunk 2 above)
Comment (furkansenharputlu) [reply core/replicated_map.go]: Also here, it is for new paragraph.

**Pull request merged**